util/request_source.go

```diff
@@ -91,6 +112,15 @@ func RequestSourceFromCtx(ctx context.Context) string {
 	return SourceUnknown
 }
 
+// IsBackgroundFromCtx checks whether the request is a background job.
+func IsBackgroundFromCtx(ctx context.Context) bool {
```

Comment:
When reviewing the code, I noticed `WithResouceGroupName` missed a `r`, can u fix it BTW? :)